ui/pages/multichain-accounts/wallet-details/wallet-details.component.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState, useMemo } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  AccountGroupId,
+  AccountWalletId,
+} from '@metamask/account-tree-controller';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+} from '../../../components/component-library';
+import {
+  AlignItems,
+  IconColor,
+  Display,
+  TextVariant,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import {
+  Content,
+  Header,
+  Page,
+} from '../../../components/multichain/pages/page';
+import { getMetaMaskHdKeyrings } from '../../../selectors';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getWalletsWithAccounts } from '../../../selectors/multichain-accounts/account-tree';
+import { getSeedPhraseBackedUp } from '../../../ducks/metamask/metamask';
+import WalletDetailsAccountItem from '../../../components/multichain/multichain-accounts/wallet-details-account-item/wallet-details-account-item';
+import UserPreferencedCurrencyDisplay from '../../../components/app/user-preferenced-currency-display/user-preferenced-currency-display.component';
+import SRPQuiz from '../../../components/app/srp-quiz-modal';
+import { setAccountDetailsAddress } from '../../../store/actions';
+
+type AccountBalance = {
+  [key: string]: string | number;
+};
+
+const WalletDetails = () => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { id } = useParams();
+  const decodedId = decodeURIComponent(id as string);
+  const walletsWithAccounts = useSelector(getWalletsWithAccounts);
+  const seedPhraseBackedUp = useSelector(getSeedPhraseBackedUp);
+  const hdKeyrings = useSelector(getMetaMaskHdKeyrings);
+  const [srpQuizModalVisible, setSrpQuizModalVisible] = useState(false);
+  const wallet = walletsWithAccounts[decodedId as AccountWalletId];
+  if (!wallet) {
+    throw new Error(`Wallet with ID "${id}" not found`);
+  }
```

Comment:
will this error crash the wallet? I wonder if we should just display a message saying something along the lines of "Unexpected problem. The wallet with ID "${id}" could not found"